integrations/gitlab/gitlab_integration/events/hooks/project_files.py

```diff
@@ -0,0 +1,111 @@
+import typing
+from typing import Any
+
+from loguru import logger
+from gitlab.v4.objects import Project
+from gitlab_integration.events.hooks.base import ProjectHandler
+from gitlab_integration.git_integration import (
+    GitLabFilesResourceConfig,
+    GitlabPortAppConfig,
+)
+from gitlab_integration.utils import ObjectKind
+from gitlab_integration.core.utils import does_pattern_apply
+
+from port_ocean.context.ocean import ocean
+from port_ocean.context.event import event
+
+
+class ProjectFiles(ProjectHandler):
+    events = ["Push Hook"]
+    system_events = ["push"]
+
+    async def _on_hook(self, body: dict[str, Any], gitlab_project: Project) -> None:
+        before, after, ref = body.get("before"), body.get("after"), body.get("ref")
+        added_files = [
+            added_file
+            for commit in body.get("commits", [])
+            for added_file in commit.get("added", [])
+        ]
+        modified_files = [
+            modified_file
+            for commit in body.get("commits", [])
+            for modified_file in commit.get("modified", [])
+        ]
+        removed_files = [
+            modified_file
+            for commit in body.get("commits", [])
+            for modified_file in commit.get("removed", [])
+        ]
+        changed_files = list(set(added_files + modified_files))
+
+        if before is None or after is None or ref is None:
+            raise ValueError(
+                "Invalid push hook. Missing one or more of the required fields (before, after, ref)"
+            )
+
+        resource_configs = typing.cast(
+            GitlabPortAppConfig, event.port_app_config
+        ).resources
+
+        matching_resource_configs = [
+            resource_config
+            for resource_config in resource_configs
+            if (
+                resource_config.kind == ObjectKind.FILE
+                and isinstance(resource_config, GitLabFilesResourceConfig)
+            )
+        ]
+        if not matching_resource_configs:
+            logger.info(
+                "Resource kind was not found in the config mapping, please update your config mapping to include the file kind"
+            )
+            return
```

Comment:
this is ok, if nothing matched, I would change it to `debug` and will change the message, to just didn't found `file kind` to handle the push event